lib/datasets/adult.rb

```diff
@@ -0,0 +1,68 @@
+require "csv"
+
+require_relative "dataset"
+
+module Datasets
+  class Adult < Dataset
+    Record = Struct.new(
+      :age,
+      :workclass,
+      :fnlwgt,
+      :education,
+      :education_num,
```

Comment:
`n_education_years` may be better because we use `n_XXX` naming convention for "the number of XXX".